src/etc/inc/plugins.inc.d/kea.inc

```diff
@@ -45,6 +45,68 @@ function kea_services()
     return $services;
 }
 
+function kea_run()
+{
+    return [
+        'static_mapping' => 'kea_staticmap',
+    ];
+}
+
+function kea_staticmap($proto = null, $valid_addresses = true, $domain_fallback = 'not.found', $ifconfig_details = null)
+{
+    $staticmap = [];
+
+    if ($proto) {
+        $proto = (int)$proto;
+    }
+
+    if (($proto ?? 4) == 4) {
+        $keav4 = new \OPNsense\Kea\KeaDhcpv4();
+
+        if (!empty((string)$keav4->general->enabled)) {
+            $reservations = $keav4->reservations;
+            $reservation = $reservations->reservation;
+
+            foreach ($reservations->reservation->iterateItems() as $reservation) {
+                $ip_address = (string)$reservation->ip_address;
+                if (empty($ip_address) && $valid_addresses) {
+                    continue;
+                }
+                $hostname = (string)$reservation->hostname;
+                if (empty($hostname)) {
+                    $hostname = null;
+                }
+                $description = (string)$reservation->description;
+                if (empty($description)) {
+                    $description = null;
+                }
```

Comment:
this feels a bit overcomplicated, empty strings should be fine. you end up thowing a PHP warning now or later because empty() should be the first sanity check then which can mute the warning. Here it does nothing.

Reply:
> you end up thowing a PHP warning now or later because empty() should be the first sanity check then which can mute the warning

I'm not quite following, should the `empty` check be removed completely or moved somewhere else?